src/resolve/resolved-mdns.c

```diff
@@ -397,6 +406,17 @@ static int on_mdns_packet(sd_event_source *s, int fd, uint32_t revents, void *us
                                 log_debug("Got a goodbye packet");
                                 /* See the section 10.1 of RFC6762 */
                                 rr->ttl = 1;
+
+                                r = sd_event_add_time_relative(
```

Comment:
dns_cache_lookup() is only called when a packet is sent, which happens indeterminately.
RFC 6762 section 10.1 does mention that the entry should be deleted form the cache 1 second later.
This helps in keeping the cache up-to-date.
Without this, the entry would linger in cache indefinitely until the next packet is sent.
It also helps to notify the client immediately when a service sends a goodbye packet and expires (#22532).

Reply:
"the entry should be deleted form the cache 1 second later" does not mean we really need to remove the entry after 1 sec. But it is enough to drop the entry when it is requested later than 1sec.

In #22532, please call `dns_cache_prune()` before `dns_cache_lookup()` in `mdns_subscriber_lookup_cache()`.

Reply:
Since goodbye packets are unsolicited, and this logic is actually independent of the RFC6762 section 5.2 stuff implemented in #22532, I figured it would make sense to add here.